UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs

```diff
@@ -83,23 +83,13 @@ public async Task<DateTime> GetRadioNavigationalWarningsLastModifiedDateTime()
             return DateTime.MinValue;
         }
 
-        public EditRadioNavigationalWarningAdmin GetRadioNavigationalWarningById(int id)
+        public RadioNavigationalWarning GetRadioNavigationalWarningById(int id)
         {
             RadioNavigationalWarning rnwWarning = _context.Set<RadioNavigationalWarning>().Find(id);
-            EditRadioNavigationalWarningAdmin rnw = new();
-            rnw.Id = rnwWarning.Id;
-            string warningName = _context.WarningType.FirstOrDefault(x => x.Id == rnwWarning.WarningType).Name;
-            rnw.WarningTypeName = warningName;
-            rnw.Reference = rnwWarning.Reference;
-            rnw.DateTimeGroup = rnwWarning.DateTimeGroup;
-            rnw.Summary = rnwWarning.Summary;
-            rnw.Content = rnwWarning.Content;
-            rnw.ExpiryDate = rnwWarning.ExpiryDate;
-            rnw.IsDeleted = rnwWarning.IsDeleted;
-            return rnw;
+            return rnwWarning;
         }
 
-        public async Task UpdateRadioNavigationalWarning(EditRadioNavigationalWarningAdmin radioNavigationalWarningAdmin)
+        public async Task UpdateRadioNavigationalWarning(RadioNavigationalWarning radioNavigationalWarningAdmin)
         {
             RadioNavigationalWarning rnw = await _context.RadioNavigationalWarnings.FirstOrDefaultAsync(r => r.Id == radioNavigationalWarningAdmin.Id);
             rnw.WarningType = radioNavigationalWarningAdmin.WarningType;
```

Comment:
No need to explicitly map values again. Please directly call update on DTO. Replace this block as mentioned below:

public async Task UpdateRadioNavigationalWarning(RadioNavigationalWarning radioNavigationalWarning)
        {
            radioNavigationalWarning.LastModified = DateTime.UtcNow;            
            _context.Update(radioNavigationalWarning);
            await _context.SaveChangesAsync();
        }

Reply:
changes done